lib/ecto/query/builder.ex

```diff
@@ -237,6 +237,11 @@ defmodule Ecto.Query.Builder do
         {field, params_acc} = escape(field, type, params_acc, vars, env)
         {{:{}, [], [:json_extract_path, [], [field, path]]}, params_acc}
 
+      {:field, _, _} ->
+        path = escape_json_path(path)
+        {field, params_acc} = escape(field, type, params_acc, vars, env)
+        {{:{}, [], [:json_extract_path, [], [field, path]]}, params_acc}
+
```

Comment:
Since the logic is duplicated maybe something like this?

```elixir
def escape({:json_extract_path, _, [field, path]} = expr, type, params_acc, vars, env) do
  field = json_field!(field)
  path = escape_json_path(path)
  {field, params_acc} = escape(field, type, params_acc, vars, env)
  {{:{}, [], [:json_extract_path, [], [field, path]]}, params_acc}
end

defp json_field({{:., _, _}, _, _} = field), do: field
defp json_field({:field, _, _} = field), do: field
defp json_field(field), do: error!("`#{Macro.to_string(field)}` is not a valid json field")
```

Reply:
I was reluctant to add it to the documentation because in my mind it was already expected for this to work, but I guess documentation is never too much :)

And nice refactoring suggestion, thanks!